app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java

```diff
@@ -348,67 +348,74 @@ public void onCopyWikicodeClicked(){
 
     @OnClick(R.id.nominateDeletion)
     public void onDeleteButtonClicked(){
-        if(AccountUtil.getUserName(getContext()).equals(media.getCreator())){
-        final ArrayAdapter<String> languageAdapter = new ArrayAdapter<>(getActivity(),
-                R.layout.simple_spinner_dropdown_list, reasonList);
-        final Spinner spinner = new Spinner(getActivity());
-        spinner.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.WRAP_CONTENT));
-        spinner.setAdapter(languageAdapter);
-        spinner.setGravity(17);
-
-        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
-        builder.setView(spinner);
-        builder.setTitle(R.string.nominate_delete)
-                .setPositiveButton(R.string.about_translate_proceed, (dialog, which) -> onDeleteClicked(spinner));
-        builder.setNegativeButton(R.string.about_translate_cancel, (dialog, which) -> dialog.dismiss());
-        AlertDialog dialog = builder.create();
-        dialog.show();
-        if(isDeleted) {
-            dialog.getButton(AlertDialog.BUTTON_POSITIVE).setEnabled(false);
-        }
-        }
-        //Reviewer correct me if i have misunderstood something over here
-        //But how does this  if (delete.getVisibility() == View.VISIBLE) {
-        //            enableDeleteButton(true);   makes sense ?
-        else{
-        AlertDialog.Builder alert = new AlertDialog.Builder(getActivity());
-        alert.setMessage(getString(R.string.dialog_box_text_nomination,media.getDisplayTitle()));
-        final EditText input = new EditText(getActivity());
-        alert.setView(input);
-        input.requestFocus();
-        alert.setPositiveButton(R.string.ok, (dialog1, whichButton) -> {
-            String reason = input.getText().toString();
-            onDeleteClickeddialogtext(reason);
-        });
-        alert.setNegativeButton(R.string.cancel, (dialog12, whichButton) -> {
-        });
-        AlertDialog d = alert.create();
-        input.addTextChangedListener(new TextWatcher() {
-            private void handleText() {
-                final Button okButton = d.getButton(AlertDialog.BUTTON_POSITIVE);
-                if (input.getText().length() == 0) {
-                    okButton.setEnabled(false);
-                } else {
-                    okButton.setEnabled(true);
+        if (AccountUtil.getUserName(getContext())!= null) {
+            if (AccountUtil.getUserName(getContext()).equals(media.getCreator())) {
```

Comment:
I would just merge these 2 if statements (if it is valid that a non logged in user can nominate things for deletion), also the null check is not in style, no space on the left hand side

Reply:
@macgills  Done. :)